internal/app/oauth/storage.go

```diff
@@ -0,0 +1,337 @@
+package oauth
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"errors"
+	"math/big"
+	mRand "math/rand"
+	"net"
+	"time"
+
+	"github.com/caos/oidc/pkg/oidc"
+	"github.com/caos/oidc/pkg/op"
+	"github.com/oklog/ulid"
+	"gopkg.in/square/go-jose.v2"
+)
+
+var (
+	appConfig *AuthSrvConfig
+	clients   map[string]op.Client
+	requests  map[string]AuthRequest
+	keySet    jose.JSONWebKeySet
+)
```

Comment:
```suggestion
```

This code works, but it is very hard to understand the code because some functions modify global variables. 

Remember that **MODIFYING GLOBAL VARIABLES IS EVIL** in 99% cases. 

In golang, the special case is modifying them only in `func init()` (automatically called when the app is initialized).

Better code is like this:

```go
type AuthStorage struct { // note: I suggest renaming AuthStorage Storage
	clients   map[string]op.Client // came from the global variables
	requests  map[string]AuthRequest // came from the global variables
	keySet    jose.JSONWebKeySet // came from the global variables
    domain    string // came from authSrvConfig
    debug     bool // came from authSrvConfig
	key       *rsa.PrivateKey
}
```